sklearn/mixture/tests/test_gmm.py

```diff
@@ -121,30 +125,33 @@ def test_lvmpdf_full_cv_non_positive_definite():
 
 
 def test_GMM_attributes():
-    n_components, n_features = 10, 4
-    covariance_type = 'diag'
-    g = mixture.GMM(n_components, covariance_type, random_state=rng)
-    weights = rng.rand(n_components)
-    weights = weights / weights.sum()
-    means = rng.randint(-20, 20, (n_components, n_features))
-
-    assert_true(g.n_components == n_components)
-    assert_true(g.covariance_type == covariance_type)
-
-    g.weights_ = weights
-    assert_array_almost_equal(g.weights_, weights)
-    g.means_ = means
-    assert_array_almost_equal(g.means_, means)
-
-    covars = (0.1 + 2 * rng.rand(n_components, n_features)) ** 2
-    g.covars_ = covars
-    assert_array_almost_equal(g.covars_, covars)
-    assert_raises(ValueError, g._set_covars, [])
-    assert_raises(ValueError, g._set_covars,
-                  np.zeros((n_components - 2, n_features)))
-
-    assert_raises(ValueError, mixture.GMM, n_components=20,
-                  covariance_type='badcovariance_type')
+    # This function tests the deprecated old GMM class
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", DeprecationWarning)
```

Comment:
Ok I've changed it in that PR but maybe it could be interested to do another PR with that.